### ISSUE TITLE
How to return custom content-type like "text/csv" as a downloadable source in browser.

### ISSUE THREAD

**Issue by weixunter** [metosin/compojure-api#235]
Trying to expose a string/ stream to a .csv file or plain text file through the browser. I am using compojure-api 1.1.2. When I tried to use Content-Disposition, with attachment, it works that I get to download a file, but the file downloaded is of .json. Also the response header looks like this:

```
{
  "date": "Wed, 29 Jun 2016 01:19:41 GMT",
  "content-type": "application/json; charset=utf-8",
  "server": "Jetty(7.6.13.v20130916)",
  "cache-control": "no-cache",
  "content-disposition": "attachment; filename=str.csv",
  "content-length": "13"
}
```

handler:
```
(defn download-file
  []
  (let [str "123,456,789"]
    {:status  200
     :headers {"Content-Type"        "text/csv"
               "Content-Length"      (str (.length str))
               "Cache-Control"       "no-cache"
               "Content-Disposition" (str "attachment; filename=str.csv")}
     :body    str}))
```

api routes swagger:
```
(defroutes app-routes
           (swagger-routes
             {:ui   "/"
              :spec "/swagger.json"
              :data {:basePath     "/"
                     :consumes     ["application/json" "text/csv"]
                     :produces     ["application/json" "text/csv"]}})

(context "/downloadfile" []
         (GET "/" []
              :produces "text/csv"
              :responses {200 {:schema s/Str
                               :description "OK"}}
              (download-file))))

(def app
  (-> (api app-routes)))
```

**Comment by weixunter** [comment]
Do I have to manually add a format to coerce the content-type? I know behind the scene compojure-api is automatically matching the response to the best content-type. In this case what can I do to force it not to coerce to application/json content-type.

**Comment by weixunter** [comment]
Closing this. 

Probably IDE issues. When running ring server-headless and trying to access the endpoint from localhost, I was able to recreate this scenario multiple time.

I got the first success after setting {:format {:formats []}} with empty []:
```
(def app
   (-> {:format {:formats []}}
       (api app-routes)))
```
Then I switched back to what I have before, and it still works. 

So I am really bumped. Using Cursive on IntelliJ and it is known that sometime I had to restart the IDE to resolve dependency.
